server/channels/product/api.go

```diff
@@ -207,9 +207,9 @@ type SystemService interface {
 //
 // The service shall be registered via app.PreferencesKey service key.
 type PreferencesService interface {
-	GetPreferencesForUser(userID string) (model.Preferences, *model.AppError)
-	UpdatePreferencesForUser(userID string, preferences model.Preferences) *model.AppError
-	DeletePreferencesForUser(userID string, preferences model.Preferences) *model.AppError
+	GetPreferencesForUser(c request.CTX, userID string) (model.Preferences, *model.AppError)
+	UpdatePreferencesForUser(c request.CTX, userID string, preferences model.Preferences) *model.AppError
+	DeletePreferencesForUser(c request.CTX, userID string, preferences model.Preferences) *model.AppError
```

Comment:
Heads up, these changes will conflict with https://github.com/mattermost/mattermost/pull/25669.

Reply:
Thanks for the heads up. I'll see if I can avoid conflicts by reverting these changes and deleting the corresponding lines in `app/preferences.go`. Git might be able to resolve the changes then